utils/python/CIME/system_test.py

```diff
@@ -489,24 +496,27 @@ def _xml_phase(self, test):
                             string = "ROOTPE_" + component_class
                             envtest.set_test_parameter(string, str(0))
                         opti_thrds = 1
-                    else:
-                        match =  re.match('P([0-9]+)x([0-9]+)', opt)
-                        if match:
-                            opti_tasks = match.group(1)
-                            opti_thrds = match.group(2)
-                            for component_class in component_classes:
-                                if component_class == "DRV":
-                                    component_class = "CPL"
-                                string = "NTASKS_" + component_class
-                                envtest.set_test_parameter(string, opti_tasks)
-                                string = "NTHRDS_" + component_class
-                                envtest.set_test_parameter(string, opti_thrds)
-                                string = "ROOTPE_" + component_class
-                                envtest.set_test_parameter(string, str(0))
-                    expect(opti_tasks is not None, "No match found for PE option %s"%opt)
-                    logger.debug (" NTASKS_xxx set to %s" %opti_tasks)
-                    logger.debug (" NTHRDS_xxx set to %s" %opti_thrds)
-                    logger.debug (" ROOTPE_xxx set to %s 0")
+                    elif match2:
+                        opti_tasks = match2.group(1)
+                        opti_thrds = match2.group(2)
+                        for component_class in component_classes:
+                            if component_class == "DRV":
+                                component_class = "CPL"
+                            string = "NTASKS_" + component_class
+                            envtest.set_test_parameter(string, opti_tasks)
+                            string = "NTHRDS_" + component_class
+                            envtest.set_test_parameter(string, opti_thrds)
+                            string = "ROOTPE_" + component_class
+                            envtest.set_test_parameter(string, str(0))
+                    elif match3:
+
+                        # handled by create_newcase
+                        continue
+                    if not match3:
```

Comment:
I think that I have it correct - I want to run the expect code in any case except when match3 is true.